mdal/api/mdal.h

```diff
@@ -122,11 +125,13 @@ MDAL_EXPORT const char *MDAL_DR_filters( DriverH driver );
 //! Caller must free memory with MDAL_CloseMesh() afterwards
 MDAL_EXPORT MeshH MDAL_LoadMesh( const char *meshFile );
 
-MDAL_EXPORT void MDAL_SaveMesh( MeshH mesh, const char *meshFile );
-
-
 //! Closes mesh, frees the memory
 MDAL_EXPORT void MDAL_CloseMesh( MeshH mesh );
+
+//! Save mesh on a file whith the specified driver. On error see MDAL_LastStatus for error type.
+//! Save the datasets if the driver has the capability
```

Comment:
I think the datasets are not saved in your implementation?

Reply:
Does 2DM format support datasets ?

Reply:
no, 2dm is mesh-only . datasets is XMDF or dat files for TUFLOW

Reply:
So, as I have only implemented the save functionality for 2DM, there is no need  implementation of datasets saving for 2DM format. no ?

Reply:
Mesh (the MeshH structure) is a) mesh definition (vertices and faces) and b) datasets ... so ... If I have loaded mesh with datasets, how do I store it...

Reply:
If the user wants to store the datasets, he has to choose a format that support the datasets. If 2DM doesn't support datasets, he has to choose another format ... As only the 2DM saving capability is implement right now, it is not possible for the moment.
The aim of this PR is to implement 2D format ...

Reply:
I am fine with implementation. Just please add to docs that it does not saves datasets, only mesh definition/frame

Reply:
I purpose :
```
//! Saves mesh on a file with the specified driver. On error see MDAL_LastStatus for error type.
//! Saves the datasets if the driver has the capability (as, right now, only the 2DM format is implemented for saving,
//! it does not save the datasets except the terrain elevation of the mesh (Z value of vertices)).
```

Reply:
I think we are not on the same page here. I propose that this function only and ever saves only mesh frame. And if you want to save datasets, you call different function to save datasets. This is because you can have a mesh that consists of 2dm frame, but XMDF datasets or any other driver combinations...

Reply:
Ok, I understand.
But, don't you think it is interesting to have the possibility to save the mesh structure and the datasets in the same file if a driver can't support this ? 

But I don't know enough each format to know if such format is present .... 

In all cases, you are right, implement a function who saves only the mesh structure is interesting. The function who saves mesh+datasets could be a distinct function and could be implemented later.
```

//! Saves mesh (only mesh structure) on a file with the specified driver. On error see MDAL_LastStatus for error type.

```